### PR TITLE
unwrap bl children when passed to append() & support bl in constructor

### PR DESCRIPTION
**Pull request by rvagg**
fixes: https://github.com/rvagg/bl/issues/26

Another improvement on this would be to move the array checking stuff down in to `append()` so even it will take arrays of stuff like the constructor.

/R=@mcollina

### CODE REVIEW

File: bl.js [hunk 1]
```diff
@@ -49,16 +47,21 @@ BufferList.prototype._offset = function (offset) {
...
+    return
+  }
 
   // coerce number arguments to strings, since Buffer(number) does
   // uninitialized memory allocation
   if (typeof buf == 'number')
     buf = buf.toString()
 
-  this._bufs.push(isBuffer ? buf : new Buffer(buf))
-  this.length += buf.length
```
Comment (rvagg) [inline bl.js]: pretty sure there was a bug in here if you pass in a string with some unicode multibyte, in fact we don't really do any unicode testing so who knows what's lurking in here

File: bl.js [hunk 2]
```diff
@@ -49,16 +47,21 @@ BufferList.prototype._offset = function (offset) {
 }
 
 BufferList.prototype.append = function (buf) {
-  var isBuffer = Buffer.isBuffer(buf) ||
-                 buf instanceof BufferList
+  if (buf instanceof BufferList) {
+    // unwrap argument into individual BufferLists
+    for (var i = 0; i < buf._bufs.length; i++)
+      this.append(buf._bufs[i])
+    return
```
Comment (mcollina) [inline bl.js]: this should be `return this`

**Comment by mcollina** [comment]
I also did https://github.com/rvagg/bl/pull/27 :).

This is ok anyway!

File: bl.js (hunk 1 above)
Comment (mcollina) [inline bl.js]: I agre!

**Comment by rvagg** [comment]
pushed an update, fixed that `return` bug, introduced more tests for that, made it so the constructor and `append()` accept the same types of arguments (sans callback) so everything is nice and symetrical and everything unrolls nicely.

**Comment by rvagg** [comment]
added some notes to the readme, rebased on 1.0.3

I propose we do a 1.1.0 with this

**Comment by jcrugzz** [comment]
This looks good :+1:. `1.1.0` sounds reasonable as it introduces more capability to the `append` api.

**Comment by mcollina** [comment]
:+1: for it being 1.1.0.

**Comment by rvagg** [comment]
done!

**Comment by rvagg** [comment]
OK, I actually messed up that merge .. so I've had to publish 1.1.1, I've unpublished 1.1.0 cause it's really just 1.0.3.

**Pull request closed without merging**